src/main/java/uk/gov/hmcts/reform/cwrdapi/service/impl/CaseWorkerServiceFacadeImpl.java

```diff
@@ -199,52 +196,56 @@ private CaseWorkerFileCreationResponse createResponse(int noOfUploadedRecords,
+                .detailedMessage(detailedMessage).errorDetails(jsrFileErrors).build();
         } else {
-            String suspendedRecordMessage = getSuspendedErrorMessage(isCaseWorker,
-                suspendedRow, false);
+            StringJoiner detailedMessage = new StringJoiner(SPACE + AND + SPACE);
+            //get the uploaded records excluding suspended records
+            int noOfUploadedRecords = totalRecords - suspendedRecords;
 
+            if (noOfUploadedRecords > 0) {
+                detailedMessage.add(format(RECORDS_UPLOADED, noOfUploadedRecords));
+            }
+            if (suspendedRecords > 0) {
+                detailedMessage.add(format(RECORDS_SUSPENDED, suspendedRecords));
+            }
             return caseWorkerFileCreationResponseBuilder
                 .message(REQUEST_COMPLETED_SUCCESSFULLY)
-                .detailedMessage(recordsUploadMessage(noOfUploadedRecords, false)
-                    + suspendedRecordMessage).build();
+                .detailedMessage(detailedMessage.toString()).build();
         }
     }
 
-    private String recordsUploadMessage(int size, boolean isPartialSuccess) {
-        String message = EMPTY;
-        if (size > 0) {
-            message = isPartialSuccess ? message.concat(DELIMITER_COMMA + SPACE) : message;
-            return message + format(RECORDS_UPLOADED, size);
+    private String constructDetailedMessage(int totalRecords, int suspendedRecords, Map<String,
+            List<ExceptionCaseWorker>> failedRecords) {
+        String detailedMessage = format(RECORDS_FAILED, failedRecords.size());
+        //get the uploaded records excluding failed records
+        int uploadedRecords = totalRecords - failedRecords.size();
+        if (suspendedRecords > 0) {
+            //sometimes the suspended records might have other failures
+            //so we need to exclude them as it goes under failed records
+            suspendedRecords = calculateFinalSuspendedRecords(failedRecords);
+            //get the uploaded records excluding suspended records
+            uploadedRecords = uploadedRecords - suspendedRecords;
         }
-        return message;
-    }
-
-    private String getSuspendedErrorMessageForJsr(boolean isCaseWorker,
-                                                  Map<String, List<ExceptionCaseWorker>> failedRecords) {
-        if (isNotEmpty(caseWorkerProfileConverter.getSuspendedRowIds())) {
-            int suspendedFailed = caseWorkerProfileConverter.getSuspendedRowIds().stream()
-                .filter(s -> failedRecords.containsKey(Long.toString(s))).collect(toList()).size();
-            return getSuspendedErrorMessage(isCaseWorker,
-                caseWorkerProfileConverter.getSuspendedRowIds().size()
-                    - suspendedFailed, true);
+        if (uploadedRecords > 0 && suspendedRecords > 0) {
+            detailedMessage = format(RECORDS_FAILED, failedRecords.size()) + DELIMITER_COMMA + SPACE
```

Comment:
format(RECORDS_FAILED, failedRecords.size()) + DELIMITER_COMMA + SPACE is common we can refactor.

Reply:
Sorry, I could not see any common code to refactor here